frontend/src/shared/hooks/filter-data.ts

```diff
@@ -0,0 +1,13 @@
+export interface UseFilterDataHook<T> {
+    filteredData: T;
+}
+
+export const useFilterData = <T>(data: T[], search: string, props?: string[]): UseFilterDataHook<T[]> => {
+    if (!Array.isArray(data) || data.length === 0) {
+        return { filteredData: data };
+    }
+
+    props = props || Object.keys(data[0]);
```

Comment:
Maybe we should give a more suitable name to the "props" variable, feels too general ? Something like "filterByKeys" or "filterByProps" maybe ?

Reply:
good point